AP/Lab_2/students.py

```diff
@@ -0,0 +1,167 @@
+from employee import Employee
+#merge --squash
+
+class Student:
+    __students_list = []
+    __HIGH_SCHOLARSHIP = 1200.0
+    __MEDIUM_SCHOLARSHIP = 1000.0
+
+    def log_creation(func):
+        def wrapper(self, *args):
+            print(f"Creating a new Student with arguments: {args}")
+            return func(self, *args)
+        return wrapper
+
+    @staticmethod
+    def __validate_grade(grade):
+        return (0 < grade <= 5)
+
+    @staticmethod
+    def __validate_course(course):
+        return 0 < course <= 5
+
+    @log_creation
+    def __init__(self, name=None, surname=None, faculty=None, course=1, min_grade=1): 
+        self._name = name if name else ""
+        self._surname = surname if surname else ""
+        self.__faculty = faculty if faculty else ""
+        self.__course = course if Student.__validate_course(course) else None
+        self.__min_grade = min_grade if Student.__validate_grade(min_grade) else None
+        if self.__min_grade is None:
+            print(f"Error: Can't add student {self._name} {self._surname}, minimum exam's grade is out of range (1-5).")
+        if self.__course is None:
+            print(f"Error: Can't add student {self._name} {self._surname}, course is out of range (1-5).")
+        else:
+            Student.__students_list.append(self)
+            print(f"Student {self._name} {self._surname} was added.")
+        self.__scholarship = 0.0
+
+    @property
+    def name(self):
+        return self._name
+    
+    @property
+    def surname(self):
+        return self._surname
+    
+    @property
+    def faculty(self):
+        return self.__faculty
+
+    @property
+    def course(self):
+        return self.__course  
+
+    @property 
+    def min_grade(self):
+        return self.__min_grade
+
+    @name.setter
+    def name(self, name):
+        self._name = name
+
+    @surname.setter
+    def surname(self, surname):
+        self._surname = surname
+
+    @faculty.setter
+    def faculty(self, faculty):
+        self.__faculty = faculty
+
+    @course.setter
+    def course(self, course):
+        if(Student.__validate_course(course)):
+            self.__course = course
+        else:
+            print("Error: Course is out of range(1-5).")
+
+    @min_grade.setter
+    def min_grade(self, min_grade):
+        if Student.__validate_grade(min_grade):
+            self.__min_grade = min_grade
+        else:
+            print("Error: Minimum exam's grade is out of range(1-5).")
+    
+    def __str__(self):
+        return f"Student {self._name} {self._surname} studies on {self.__course} course of {self.__faculty} faculty with minimum exam's grade {self.__min_grade} and scholarship of {self.__scholarship} grn."
+
+    @classmethod    
+    def show_all_students(cls):
+        print("\n-------List of students-------")
+        if(len(cls.__students_list) == 0):
+            print("Error: Students list is empty.")
+            return 
```

Comment:
I believe it won't print anything if list is empty, check if return here is neccessary